### ISSUE TITLE
Implement soft-delete

### ISSUE THREAD

**Issue by nblair** [sonatype-nexus-community/nexus-blobstore-google-cloud#2]
Soft deletes, e.g. `BlobStore#delete(BlobId, String)`, is current implemented as a hard delete. Refactor to meet conventional soft delete implementation.

**Comment by DarthHater** [comment]
@nblair is there perhaps some examples of how this was done with s3 or the regular blobstore that we can add to help others if they'd like to join in?

**Comment by nblair** [comment]
@DarthHater yep; the [File](https://github.com/sonatype/nexus-public/blob/master/components/nexus-blobstore-file/src/main/java/org/sonatype/nexus/blobstore/file/FileBlobStore.java#L458) and [S3](https://github.com/sonatype/nexus-public/blob/master/plugins/nexus-blobstore-s3/src/main/java/org/sonatype/nexus/blobstore/s3/internal/S3BlobStore.java#L342) implementations provide a good example to follow.

**Comment by nblair** [comment]
I did a little deeper investigation on this yesterday. 

In the S3 implementation, we implemented soft-delete by adding a tag to a "soft deleted" blob. We also assigned a Lifecycle to the bucket to delete objects in the bucket with that tag older than X days.
From my understanding of the Google Cloud Storage API, it does not provide a facility to tag individual objects within the bucket like we could with the S3 implementation. 
As a result, we'll need to implement the `BlobStore#compact(BlobStoreUsageChecker)` method with this. The [File backed implementation](https://github.com/sonatype/nexus-public/blob/master/components/nexus-blobstore-file/src/main/java/org/sonatype/nexus/blobstore/file/FileBlobStore.java#L551) may serve as a guide, although I'm curious to see if storing a `QueueFile` in the bucket for each NXRM node will be applicable here.
